Maps part2 chen

MCharming98 opened the pull request:
Implemented basic logic of presenting the map with the boba shop data. Feel free to open boba-shops.html and play around!

MCharming98 committed:
Add source csv file for display on map

The file contains data about the boba places in the bay area.
Source: https://www.kaggle.com/vnxiclaire/bobabayarea
License: CCO: Public Domain

MCharming98 committed:
Add source csv file for display on map

The file contains data about the boba places in the bay area.
Source: https://www.kaggle.com/vnxiclaire/bobabayarea
License: CCO: Public Domain

MCharming98 committed:
Add servlet to read data from csv file into json array

MCharming98 committed:
Add servlet to read data from csv file into json array

MCharming98 committed:
Add html page to present fetched data on google maps

MCharming98 committed:
Add html page to present fetched data on google maps

MCharming98 committed:
Fixed an error in the data where the name has a comma inside

MCharming98 committed:
Fixed an error in the data where the name has a comma inside

MCharming98 committed:
Add more fields to the BobaShop class in order to present on the website

MCharming98 committed:
Add more fields to the BobaShop class in order to present on the website

MCharming98 committed:
Add html page to present the map and shop info

MCharming98 committed:
Add html page to present the map and shop info

MCharming98 committed:
Separate css and js code from html page

MCharming98 committed:
Separate css and js code from html page

MCharming98 committed:
Add bounce animation to the marker that is clicked, and changed the icon of the marker to customed one

MCharming98 committed:
Add bounce animation to the marker that is clicked, and changed the icon of the marker to customed one

MCharming98 committed:
Added image source of the marker icon

MCharming98 committed:
Added image source of the marker icon

MCharming98 committed:
Paused using CSS file as the map is not framed by style text in CSS. Added comments in boba-shop-loader.js

MCharming98 committed:
Paused using CSS file as the map is not framed by style text in CSS. Added comments in boba-shop-loader.js

MCharming98 committed:
Inline css code into html file for now

MCharming98 committed:
Inline css code into html file for now

MCharming98 committed:
Separate css style from the html file

MCharming98 committed:
Separate css style from the html file

MCharming98 committed:
Fix lint errors

MCharming98 committed:
Fix lint errors

MCharming98 committed:
Fix lint errors

MCharming98 committed:
Fix lint errors

MCharming98 committed:
Fix lint errors

MCharming98 committed:
Fix lint errors

MCharming98 committed:
Fixing lint error

MCharming98 committed:
Fixing lint error

MCharming98 committed:
Fixing lint errors

MCharming98 committed:
Fixing lint errors

MCharming98 committed:
Fixing lint errors

MCharming98 committed:
Fixing lint errors

MCharming98 committed:
Merge branch 'maps_part2_Chen' of https://github.com/maksymko/codeu-starter-project into maps_part2_Chen

MCharming98 commented:
I got a pretty weird thing: So as I was updating my branch I noticed it says my branch is out of date with master. So I did git fetch, rebase ... and I found myself resolving merge conflict with the changes I made earlier. I don't know what happens but I know this is not right :( So I pulled my own branch and pushed it and... seems updated with master??

MCharming98 committed:
Fixing lint errors

MCharming98 committed:
Fixing lint errors

MCharming98 committed:
Clean up some files:

boba-shops.html - Moved last style code into css
boba-shop.css - Add more one style rule, removed extra whitespace
boba-shop-loader.js - Add comments, removed extra whitespace

MCharming98 committed:
Remove extra whitespace

maksymko commented on src/main/java/com/google/codeu/servlets/BobaDataServlet.java:
```diff
@@ -0,0 +1,80 @@
+  /**
+   * Read and parse data from the csv file into JsonArray.
+   */
+  private JsonArray parseData() {
```
"parseData" is a vague name. What data? Where does it come from?

I would suggest splitting this function into two:

1. Static "parseCSV", which would take Scanner and return JsonArray.
2. getCSV or getStaticCSV, which would load the file and return scanner.

maksymko commented on src/main/java/com/google/codeu/servlets/BobaDataServlet.java:
```diff
@@ -0,0 +1,80 @@
+    Gson gson = new Gson();
+
+    Scanner scanner = 
+        new Scanner(getServletContext().getResourceAsStream("/WEB-INF/bayarea_boba_spots.csv"));
```
I think it's better to set the path to the file in private constant.

maksymko commented on src/main/webapp/js/boba-shop-loader.js:
```diff
@@ -0,0 +1,53 @@
+  }).then((bobaShops) => {
+      
+    const map = new google.maps.Map(document.getElementById('map'), {
+      center: {lat: 37.7749, lng: -122.4194}, //San Francisco
```
You can document this with code:
`const SanFrancisco = {lat: ..., lng: ...}`

MCharming98 committed:
Refactored some code:

BobaDataServlet.java: Separate csv reading functionality from the parsing method into its own method.
boba-shop-loader.js: Put the coordinate of San Francisco into const variable.

MCharming98 committed:
Changed method names to fix lint errors

maksymko commented on src/main/java/com/google/codeu/servlets/BobaDataServlet.java:
```diff
@@ -0,0 +1,90 @@
+    bobaShopsArray = new JsonArray();
+    Gson gson = new Gson();
+
+    Scanner scanner = getDataFile(path);
```
The idea of the change that I've suggested was to clearly separate the responsibilities of *loading* data from *parsing* data. Right now `parseDataFile` still does both: it loads the data and parses it. The fact that loading is in the separate function does not change much.

IMO, it would be better to have a function that takes *data* and parses it. Scanner looks like a good candidate. So, for example, then in your `init()` method you would call

```
Scanner data_scanner = loadBobaData();
this.bobaShopsArray = parseCSV();
```
or something like that.
The idea is that if later you decide to load the data from somewhere else, you can leave `parseCSV` function the same and would only need to change `init()` so it would call different function for loading data. This is what it means to have flexible API.

I'll leave it up to you to decide whether to make this change or not.

MCharming98 committed:
Separated data parsing functionality from data reading.
